src/AbstractPolygon.jl

```diff
@@ -44,3 +47,127 @@ The ambient dimension of the polygon, which is 2.
 @inline function dim(P::AbstractPolygon)::Int
     return 2
 end
+
+"""
+    jump2pi(x::N)::N where {N<:AbstractFloat}
+
+Return ``x + 2π`` if ``x`` is negative, otherwise return ``x``.
+
+### Input
+
+- `x` -- real scalar
+
+### Output
+
+``x + 2π`` if ``x`` is negative, ``x`` otherwise.
+
+### Examples
+
+```jldoctest
+julia> jump2pi(0.0)
+0.0
+julia> jump2pi(-0.5)
+5.783185307179586
+julia> jump2pi(0.5)
+0.5
+```
+"""
+function jump2pi(x::N)::N where {N<:AbstractFloat}
+    x < zero(N) ? 2 * pi + x : x
+end
+
+"""
+    <=(u::AbstractVector{N}, v::AbstractVector{N})::Bool where {N<:AbstractFloat}
+
+Compares two 2D vectors by their direction.
+
+### Input
+
+- `u` --  first 2D direction
+- `v` --  second 2D direction
+
+### Output
+
+True iff ``\\arg(u) [2π] ≤ \\arg(v) [2π]``
+
+### Notes
+
+The argument is measured in counter-clockwise fashion, with the 0 being the
+direction (1, 0).
+
+### Algorithm
+
+The implementation uses the arctangent function with sign, `atan2`.
+"""
+function <=(u::AbstractVector{N},
+            v::AbstractVector{N})::Bool where {N<:AbstractFloat}
+    return jump2pi(atan2(u[2], u[1])) <= jump2pi(atan2(v[2], v[1]))
+end
+
+"""
+    quadrant(w::AbstractVector{N})::Int where {N<:Real}
+
+Compute the quadrant where the direction `w` belongs.
+
+### Input
+
+- `w` --  direction
+
+### Output
+
+An integer from 0 to 3, with the following convention:
+
+     ^
+   1 | 0
+  ---+-->
+   2 | 3
+
+### Algorithm
+
+This function is inspired from AGPX's answer for C++ in the stackoverflow question
+[Sort points in clockwise order?](https://stackoverflow.com/questions/6989100/sort-points-in-clockwise-order).
```

Comment:
You can refer to the exact post by clicking the "share" button (it is quite hidden):
https://stackoverflow.com/questions/6989100/sort-points-in-clockwise-order/46635372#46635372

Reply:
good finding!